core/src/server/helpers.rs

```diff
@@ -224,7 +274,7 @@ impl MethodResponse {
 	pub fn error<'a>(id: Id, err: impl Into<ErrorObject<'a>>) -> Self {
 		let err = ResponsePayload::error_borrowed(err);
 		let result = serde_json::to_string(&Response::new(err, id)).expect("JSON serialization infallible; qed");
-		Self { result, success: false }
+		Self { result, success_or_error: MethodResponseResult::Failed(ErrorCode::InternalError.code()) }
```

Comment:
maybe a dumb question but why does this set an error code which is different from the error code in the `ErrorObject`?

Reply:
it's a bug, good catch